service/matching/handler/engine.go

```diff
@@ -515,7 +515,8 @@ pollLoop:
 		task, err := e.getTask(pollerCtx, taskListID, nil, taskListKind)
 		if err != nil {
 			// TODO: Is empty poll the best reply for errPumpClosed?
-			if errors.Is(err, tasklist.ErrNoTasks) || errors.Is(err, errPumpClosed) {
+			var taskListNotOwnedByHostError *cadence_errors.TaskListNotOwnedByHostError
```

Comment:
Lets separate cases by error types.
I don't like that we have a TODO that now does not cover the if statement

Reply:
Reverted this change - we just return the error and retry on frontend